From 3c3619d325f7b95fd104f374b89ec1c789adf061 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Mon, 8 Dec 2014 16:22:55 +0200
Subject: [PATCH 05/13] Camera2: Remove settings preferences only once

Browsing through camera settings crashes on devices with only
a back- or a front facing camera. Fix NPE on removing preferences
that are already removed.

Change-Id: I00b4549e18ea44975bf795da9384892d8ef2ee16
---
 src/com/android/camera/settings/CameraSettingsActivity.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/settings/CameraSettingsActivity.java b/src/com/android/camera/settings/CameraSettingsActivity.java
index 50641fd22..32afbed49 100644
--- a/src/com/android/camera/settings/CameraSettingsActivity.java
+++ b/src/com/android/camera/settings/CameraSettingsActivity.java
@@ -147,6 +147,7 @@ public class CameraSettingsActivity extends FragmentActivity {
         private String mPrefKey;
         private boolean mHideAdvancedScreen;
         private boolean mGetSubPrefAsRoot = true;
+        private boolean mPreferencesRemoved = false;
 
         // Selected resolutions for the different cameras and sizes.
         private PictureSizes mPictureSizes;
@@ -285,18 +286,19 @@ public class CameraSettingsActivity extends FragmentActivity {
         private void setVisibilities() {
             PreferenceGroup resolutions =
                     (PreferenceGroup) findPreference(PREF_CATEGORY_RESOLUTION);
-            if (mPictureSizes.backCameraSizes.isEmpty()) {
+            if ((mPictureSizes.backCameraSizes.isEmpty()) && !mPreferencesRemoved) {
                 recursiveDelete(resolutions,
                         findPreference(Keys.KEY_PICTURE_SIZE_BACK));
                 recursiveDelete(resolutions,
                         findPreference(Keys.KEY_VIDEO_QUALITY_BACK));
             }
-            if (mPictureSizes.frontCameraSizes.isEmpty()) {
+            if ((mPictureSizes.frontCameraSizes.isEmpty()) && !mPreferencesRemoved) {
                 recursiveDelete(resolutions,
                         findPreference(Keys.KEY_PICTURE_SIZE_FRONT));
                 recursiveDelete(resolutions,
                         findPreference(Keys.KEY_VIDEO_QUALITY_FRONT));
             }
+            mPreferencesRemoved = true;
         }
 
         /**
-- 
2.17.1

